routers/web/repo/view.go

```diff
@@ -1168,3 +1177,31 @@ func Forks(ctx *context.Context) {
 
 	ctx.HTML(http.StatusOK, tplForks)
 }
+
+func loadPinData(ctx *context.Context) error {
```

Comment:
maybe we can cache pin data

Reply:
I see a good reason for caching that data, but I'm not confident in implementing this myself. I'm not too accustomed to this codebase and, since some of the pin data can be sensitive data (since some pinned repos can be private), that would be a hard task to do without any guidance.

Reply:
I would ask @lunny.

Reply:
I think it's unnecessary to cache these pinned repositories.